app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/Session.kt

```diff
@@ -1,287 +1,162 @@
+    val links: String = "", // Comma separated Markdown formatted links, see ParserTask#parseFahrplan.
+    val url: String = "",
+    @Deprecated(
+        "The value of this field is generated by {@link FahrplanParser} " +
+                "when parsing the schedule. It contributes to how the rooms are sorted in " +
+                "the user interface, see {@link SessionsTransformer}. But it should not be " +
+                "used by any other code!"
+    )
+    val dayIndex: Int = 0, // XML values start with 1
+    val dateText: String = "", // YYYY-MM-DD
+    val dateUTC: Long = 0, // milliseconds
+    val timeZoneOffset: ZoneOffset? = null,
+    val startTime: Int = 0, // minutes since day start
+    val relStartTime: Int = 0, // minutes since conference start
+    val duration: Int = 0, // minutes
+    val roomName: String = "",
+    val roomIdentifier: String = "", // Unique identifier of a room, e.g. "bccb6a5b-b26b-4f17-90b9-b5966f5e34d8"
+    val roomIndex: Int = 0,
+    val speakers: List<String> = emptyList(),
+    val track: String = "",
+    val type: String = "",
+    val language: String = "",
+    val slug: String = "",
+    val recordingLicense: String = "",
+    val recordingOptOut: Boolean = false,
+    val highlight: Boolean = false,
+    val hasAlarm: Boolean = false,
+
+    val changedTitle: Boolean = false,
+    val changedSubtitle: Boolean = false,
+    val changedRoomName: Boolean = false,
+    val changedDayIndex: Boolean = false,
+    val changedStartTime: Boolean = false,
+    val changedDuration: Boolean = false,
+    val changedSpeakers: Boolean = false,
+    val changedLanguage: Boolean = false,
+    val changedRecordingOptOut: Boolean = false,
+    val changedTrack: Boolean = false,
+
+    val changedIsNew: Boolean = false,
+    val changedIsCanceled: Boolean = false,
+) {
+
+    override fun equals(other: Any?) =
+        other is Session && EssentialSession(this) == EssentialSession(other)
+
+    override fun hashCode() = EssentialSession(this)
+        .hashCode()
+
+    override fun toString() = EssentialSession(this)
+        .toString()
+        .replaceFirst("EssentialSession", "Session")
 
     /**
      * Returns a moment based on the start time milliseconds.
-     * <p/>
-     * Don't use in {@link Conference.Companion#ofSessions)} as long as {@link #relStartTime} is supported.
-     * See: <a href="https://github.com/EventFahrplan/EventFahrplan/commit/5a4022b00434700274a824cc63f6d54a18b06fac">5a402</a>
+     *
+     * Don't use in [Conference.ofSessions] as long as [relStartTime] is supported.
+     * See: [5a402](https://github.com/EventFahrplan/EventFahrplan/commit/5a4022b00434700274a824cc63f6d54a18b06fac)
      */
-    public Moment getStartsAt() {
-        if (dateUTC <= 0) {
-            throw new IllegalArgumentException("Field 'dateUTC' must be more than 0.");
+    val startsAt: Moment
+        get() {
+            require(dateUTC > 0) { "Field 'dateUTC' must be more than 0." }
```

Comment:
`require()` throws an `IllegalArgumentException`, but there is no argument. Use `check()` which throws `IllegalStateException` instead.

Reply:
:heavy_check_mark: Changed now.